nexus/types/src/deployment/planning_input.rs

```diff
@@ -124,10 +126,14 @@ impl PlanningInput {
 
     pub fn target_clickhouse_zone_count(&self) -> usize {
         if let Some(policy) = &self.policy.clickhouse_policy {
-            if policy.deploy_with_standalone {
-                SINGLE_NODE_CLICKHOUSE_REDUNDANCY
-            } else {
-                0
+            match policy.mode {
```

Comment:
This might be cleaner if squished into the `if let`; untested, but something like

```rust
match self.policy.clickhouse_policy.map(|policy| policy.mode) {
    Some(ClickhouseMode::ClusterOnly { .. }) => 0,
    Some(ClickhouseMode::SingleNodeOnly)
    | Some(ClickhouseMode::Both { .. })
    | None => SINGLE_NODE_CLICKHOUSE_REDUNDANCY,
}
````

Reply:
Agreed! This looks nicer. Done in [108ec11](https://github.com/oxidecomputer/omicron/pull/6862/commits/108ec119ea0dee25281e6e1f4031d90ed8afb67a)